deployments/tekton.go

```diff
@@ -334,52 +338,87 @@ func (k Tekton) Upgrade(ctx context.Context, c *kubernetes.Cluster, ui *termui.U
 }
 
 // The equivalent of:
-// kubectl get secret -n tekton-staging registry-tls-self -o json | jq -r '.["data"]["ca"]' | base64 -d | openssl x509 -hash -noout
+// kubectl get secret -n tekton-staging epinio-registry-tls -o json | jq -r '.["data"]["ca.crt"]' | base64 -d | openssl x509 -hash -noout
 // written in golang.
-func getRegistryCAHash(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI) (string, error) {
+func getRegistryCAHash(ctx context.Context, c *kubernetes.Cluster) (string, error) {
 	secret, err := c.Kubectl.CoreV1().Secrets(TektonStagingNamespace).
-		Get(ctx, "registry-tls-self", metav1.GetOptions{})
+		Get(ctx, fmt.Sprintf("%s-tls", RegistryDeploymentID), metav1.GetOptions{})
 	if err != nil {
 		return "", err
 	}
 
-	hash, err := GenerateHash(secret.Data["ca"])
+	hash, err := GenerateHash(secret.Data["ca.crt"])
 	if err != nil {
 		return "", err
 	}
 
 	return hash, nil
 }
 
-func applyTektonStaging(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI) (string, error) {
-	caHash, err := getRegistryCAHash(ctx, c, ui)
-	if err != nil {
-		return "", errors.Wrapf(err, "Failed to get registry CA from %s namespace", TektonStagingNamespace)
-	}
-
+func applyTektonStaging(ctx context.Context, c *kubernetes.Cluster, domain string) error {
 	yamlPathOnDisk, err := helpers.ExtractFile(tektonStagingYamlPath)
 	if err != nil {
-		return "", errors.New("Failed to extract embedded file: " + tektonStagingYamlPath + " - " + err.Error())
+		return errors.New("Failed to extract embedded file: " + tektonStagingYamlPath + " - " + err.Error())
 	}
 	defer os.Remove(yamlPathOnDisk)
 
 	fileContents, err := ioutil.ReadFile(yamlPathOnDisk)
 	if err != nil {
-		return "", err
+		return err
 	}
 
-	// Constructing the name of the cert file as required by openssl.
-	// Lookup "subject_hash" in the docs: https://www.openssl.org/docs/man1.0.2/man1/x509.html
-	re := regexp.MustCompile(`{{CA_SELF_HASHED_NAME}}`)
-	renderedFileContents := re.ReplaceAll(fileContents, []byte(caHash+".0"))
+	tektonTask := &v1beta1.Task{}
+	err = yaml2.Unmarshal(fileContents, tektonTask, func(opt *json.Decoder) *json.Decoder {
+		opt.UseNumber()
+		return opt
+	})
+	if err != nil {
+		return errors.Wrapf(err, "failed to unmarshal task %s", string(fileContents))
+	}
 
-	tmpFilePath, err := helpers.CreateTmpFile(string(renderedFileContents))
+	// Add volume and volume mount of registry-certs for local deployment
+	// since tekton should trust the registry-certs
+	if strings.Contains(domain, "omg.howdoi.website") {
```

Comment:
I think a comment should explain why we handle omg.howdoi.website differently

Reply:
In general I would also add the relevant documentation from the issue to the commit messages - if it fits - but this does not have to be fixed here, just for the future.

Reply:
There is already a comment that for local deployment, we need to add volume and volume mounts so that tekton can trust them.